lib/thumbs/pull_request_worker.rb

```diff
@@ -187,45 +204,62 @@ def try_run_build_step(name, command)
           status[:output] = sanitize_text(output)
           status[:exit_code] = exit_code.to_i
 
-          @build_status[:steps][name.to_sym]=status
+          build_status[:steps][name.to_sym] = status
           debug_message "[ #{name.upcase} ] [#{result.upcase}] \"#{command}\""
           return status
         end
 
       rescue Timeout::Error => e
-        status[:ended_at]=DateTime.now
+        status[:ended_at] = DateTime.now
         status[:result] = :error
-        status[:message] = "Timeout reached (#{@timeout} seconds)"
+        status[:message] = "Timeout reached (#{config_timeout} seconds)"
         status[:output] = e
 
-        @build_status[:steps][name.to_sym]=status
+        build_status[:steps][name.to_sym] = status
         debug_message "[ #{name.upcase} ] [ERROR] \"#{command}\" #{e}"
         return status
       end
     end
 
     def events
-      client.repository_events(@repo).collect { |e| e.to_h }
+      @events ||= octokit_client.repository_events(repo).collect { |e| e.to_h }
     end
 
-    def push_time_stamp(sha)
-      time_stamp=events.collect { |e| e[:created_at] if e[:type] == 'PushEvent' && e[:payload][:head] == sha }.compact.first
-      time_stamp ? time_stamp : pr.created_at
+    def push_timestamp(sha)
+      push_event = events.detect { |e| e[:payload][:head] == sha &&
+                                  e[:type] == 'PushEvent' }
+      timestamp = event_timestamp(push_event)
+      timestamp || pull_request.created_at
     end
 
-    def comments_after_most_recent_commit
-      comments_after_sha=all_comments.compact.collect do |c|
-        comment_timestamp=DateTime.parse(c[:created_at].to_s)
-        c.to_h if comment_timestamp > most_recent_commit_timestamp
-      end.compact
+    def to_timestamp(value)
+      DateTime.parse(value.to_s)
+    end
+
+    def event_timestamp(event)
+      return unless event
+      to_timestamp(event[:created_at])
+    end
+
+    def comment_timestamp(comment)
+      to_timestamp(comment[:created_at])
+    end
+
+    def comments_since_most_recent_commit
+        @comments_since_most_recent_commit ||=
+        all_comments.compact.select { |c|
+          config_comments_since_disabled? ||
+          comment_timestamp(c) > most_recent_commit_timestamp
+        }.map { |c| c.to_h }.compact
```

Comment:
`select { ... }.map { ... }` basically is easier to reason about. The `select` filters out whatever doesn't match in an Enumerable fashion (w/o creating an array), so each matching element is fed to whatever Enumerable method is chained (in this case `map`).

I didn't aggressive attempt to remove `compact` b/c well there's more than enough changed as is.